File: src/main/java/org/jukeboxmc/event/entity/ProjectileHitEntityEvent.java
package org.jukeboxmc.event.entity;

import org.jukeboxmc.entity.Entity;
import org.jukeboxmc.entity.projectile.EntityProjectile;
import org.jukeboxmc.event.Cancellable;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class ProjectileHitEntityEvent extends EntityEvent implements Cancellable {

    private EntityProjectile projectile;

    public ProjectileHitEntityEvent( Entity entity, EntityProjectile projectile ) {
        super( entity );
        this.projectile = projectile;
    }

    public EntityProjectile getProjectile() {
        return this.projectile;
    }
}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/event/entity/ProjectileHitEntityEvent.kt
package org.jukeboxmc.api.event.entity

import org.jukeboxmc.api.entity.Entity
import org.jukeboxmc.api.entity.projectile.EntityProjectile
import org.jukeboxmc.api.event.Cancellable

class ProjectileHitEntityEvent(
    entity: Entity,
    private val entityProjectile: EntityProjectile
) : EntityEvent(entity), Cancellable {

    fun getProjectile(): EntityProjectile {
        return this.entityProjectile
    }

}